src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2328,137 +2379,221 @@ private void MeasurePrompts(int startLine, int endLine, out int minPromptLength,
                 minPromptLength = maxPromptLength = promptSpan.Length;
             }
 
-            /// <summary>Implements <see cref="IInteractiveWindowOperations.Cut"/>.</summary>
-            public void Cut()
+            /// <summary>Implements <see cref="IInteractiveWindowOperations.Backspace"/>.</summary>
+            public bool Backspace()
             {
-                if (!TextView.Selection.IsEmpty)
+                using (var transaction = UndoHistory_TestOnly.CreateTransaction(InteractiveWindowResources.Backspace))
                 {
-                    if (CutOrDeleteSelection(isCut: true))
+
+                    if (DeleteHelper(isBackspace: true))
                     {
-                        MoveCaretToClosestEditableBuffer();
+                        // don't record undo histroy for standard-input
+                        if (!ReadingStandardInput)
+                        {
+                            transaction.Complete();
+                        }
                     }
-                    return;
-                }
-
-                var caretPosition = TextView.Caret.Position.BufferPosition;
-
-                // don't cut and move caret if it's in readonly buffer (except in active prompt)
-                if (MapToEditableBuffer(caretPosition) != null ||
-                    IsInActivePrompt(caretPosition))
-                {
-                    CutOrDeleteCurrentLine(isCut: true);
-                    Home(false);
                 }
+                // DeleteHelper handles deleteion completely, 
+                // so always return true to indicate the backspace is done.
+                return true;
```

Comment:
Just looked at the comments for `InteractiveOperations`, I think it meas something was deleted. Will fix.